plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/tls/DefaultSslContextProvider.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.arrow.flight.bootstrap.tls;
+
+import org.opensearch.plugins.SecureTransportSettingsProvider;
+
+import javax.net.ssl.SSLException;
+
+import java.util.Locale;
+
+import io.grpc.netty.shaded.io.netty.handler.ssl.ApplicationProtocolConfig;
```

Comment:
Why do we need to use shaded Netty here?

Reply:
we are using grpc-netty-shaded instead of netty dependencies directly for any thing related to grpc server.

Reply:
I think i will remove grpc-netty-shaded, it might not work since we are no more shading flight jars to use grpc-netty-shaded. 
Will add it back once we will start using shaded dependencies.

Reply:
removed `grpc-netty-shaded`